qcodes/instrument/parameter.py

```diff
@@ -1130,20 +1141,49 @@ class Parameter(_BaseParameter):
     """
 
     def __init__(
-            self, name: str,
-            instrument: Optional['InstrumentBase'] = None,
-            label: Optional[str] = None,
-            unit: Optional[str] = None,
-            get_cmd: Optional[Union[str, Callable[..., Any], bool]] = None,
-            set_cmd:  Optional[Union[str, Callable[..., Any], bool]] = False,
-            initial_value: Optional[Union[float, str]] = None,
-            max_val_age: Optional[float] = None,
-            vals: Optional[Validator[Any]] = None,
-            docstring: Optional[str] = None,
-            initial_cache_value: Optional[Union[float, str]] = None,
-            **kwargs: Any) -> None:
-        super().__init__(name=name, instrument=instrument, vals=vals,
-                         max_val_age=max_val_age, **kwargs)
+        self,
+        name: str,
+        instrument: Optional["InstrumentBase"] = None,
+        label: Optional[str] = None,
+        unit: Optional[str] = None,
+        get_cmd: Optional[Union[str, Callable[..., Any], bool]] = None,
+        set_cmd: Optional[Union[str, Callable[..., Any], bool]] = False,
+        initial_value: Optional[Union[float, str]] = None,
+        max_val_age: Optional[float] = None,
+        vals: Optional[Validator[Any]] = None,
+        docstring: Optional[str] = None,
+        initial_cache_value: Optional[Union[float, str]] = None,
+        bind_to_instrument: bool = True,
+        **kwargs: Any,
+    ) -> None:
+        if instrument is not None and bind_to_instrument:
+            existing_parameter = instrument.parameters.get(name, None)
+
+            if existing_parameter:
+
+                # this check is redundant since its also in the baseclass
+                # but if we do not put it here it would be an api break
+                # as KeyErrors are now transformed into ValueErrors.
+                # do we care about that?
+                if not existing_parameter.abstract:
+                    raise KeyError(f"Duplicate parameter name {name}")
+
+                existing_unit = getattr(existing_parameter, "unit", None)
+                if existing_unit != unit:
+                    raise ValueError(
+                        f"The unit of the parameter '{name}' is '{unit}'. "
+                        f"This is inconsistent with the unit defined in the "
+                        f"base class"
+                    )
```

Comment:
if below i explain/undersatnd it correctly, then let's keep the redundancy. OR move the unit check after the super.init?
```suggestion
            if existing_parameter:

                # this check is redundant since its also in the baseclass
                # but if we do not put it here it would be an api break
                # as parameter duplication check won't be done first,
                # hence for parameters that are duplicates ANd have
                # wrong units, users will be getting ValueError where 
                # they used to have KeyError before.
                if not existing_parameter.abstract:
                    raise KeyError(f"Duplicate parameter name {name}")

                existing_unit = getattr(existing_parameter, "unit", None)
                if existing_unit != unit:
                    raise ValueError(
                        f"The unit of the parameter '{name}' is '{unit}'. "
                        f"This is inconsistent with the unit defined in the "
                        f"base class"
                    )
```

Reply:
I don't think putting after super init is good since that actually does the registration (Which would then have to be undone if unit is wrong)